### PR TITLE
fix: `@trpc/client` has undefined `global`

### LINKED ISSUE #2493: bug: v10 @trpc/client has undefined `global` in workers
### Provide environment information

```
  System:
    OS: macOS 12.5
    CPU: (8) arm64 Apple M1
    Memory: 158.41 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.5.0 - ~/.nvm/versions/node/v18.5.0/bin/node
    Yarn: 1.22.17 - ~/.yarn/bin/yarn
    npm: 8.12.1 - ~/.nvm/versions/node/v18.5.0/bin/npm
  Browsers:
    Chrome: 104.0.5112.101
    Chrome Canary: 106.0.5249.0
    Firefox Developer Edition: 104.0
    Safari: 15.6
  npmPackages:
    @trpc/client: 10.0.0-alpha.47 => 10.0.0-alpha.47 
    @trpc/server: =10.0.0-alpha.47 => 10.0.0-alpha.47 
```

### Describe the bug

When attempting to deploy my app I was getting the following error:
```
✘ [ERROR] A request to the Cloudflare API (/accounts/000000011111111122222222/workers/scripts/app-edge) failed.

  Uncaught ReferenceError: global is not defined
    at worker.js:10:10
   [code: 10021]

  
  If you think this is a bug, please open an issue at:
  https://github.com/cloudflare/wrangler2/issues/new/choose
```

### To reproduce

Try to deploy something using something akin to my gist for [wrapping Durable Objects in TRPC](https://gist.github.com/colelawrence/ba912db3cd1539ee820c3fe5e67654df)

### Additional information

I was able to fix in the short term by adding a `banner` via esbuild: `if (typeof global === "undefined") { globalThis.global = globalThis; }`

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### PR DESCRIPTION
**Pull request by KATT**
Closes #2493

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Comment by colelawrence** [comment]
Taking a look, now.

File: packages/client/src/internals/fetchHelpers.ts [hunk 1]
```diff
@@ -2,11 +2,14 @@ export function getWindow() {
   if (typeof window !== 'undefined') {
     return window;
   }
-  return global;
+  if (typeof global !== 'undefined') {
+    return global;
+  }
+  return undefined;
```
Comment (colelawrence) [inline packages/client/src/internals/fetchHelpers.ts:8]:
```suggestion
  if (typeof globalThis !== 'undefined') {
    return globalThis;
  }
  return undefined;
```

**Review by colelawrence — CHANGES_REQUESTED** [review]
Consider either adopting the shim described in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis#search_for_the_global_across_environments

Or including a search for `globalThis`, which is how I fixed it for my workers environment.

**Comment by KATT** [comment]
@colelawrence do you think we can get rid of `typeof global` altogether?

**Comment by colelawrence** [comment]
@KATT, re: getting rid of `global` check: I don't know. Perhaps add a comment like `// Hmm: Maybe checking for global variable is no longer necessary with globalThis and window checks already.`

I don't feel strongly either way.

**Review by colelawrence — APPROVED** [review]
LGTM. Thanks!

File: packages/client/src/internals/fetchHelpers.ts [hunk 2]
```diff
@@ -2,11 +2,17 @@ export function getWindow() {
   if (typeof window !== 'undefined') {
     return window;
   }
-  return global;
+  if (typeof global !== 'undefined') {
+    return global;
+  }
+  if (typeof globalThis !== 'undefined') {
```
Comment (sachinraja) [inline packages/client/src/internals/fetchHelpers.ts:8]: We don't need this function at all if we're using `globalThis`. It's supported in Node v12, Deno, and all the browsers.

Comment (colelawrence) [reply packages/client/src/internals/fetchHelpers.ts:8]: @KATT, I've reduced the complexity of this code and implemented this suggestion in PR in https://github.com/trpc/trpc/pull/2563

**Pull request closed without merging**